### DISCUSSION TITLE
Upgrading Nodes with Active Plans

### DISCUSSION THREAD

**Discussion by joesixpack** [subquery/subql#963] [Frontier TestNet - Bug Reporting]
I think there should be a process to allow for upgrading the node while active plans are in play.  Perhaps a process that lets you announce "not indexing to upgrade" that continues to service the existing agreements but prevents any more from being initiated so that at the end, it can actually be accomplished.

Since I had to force upgrade without being able to announce "not indexing", I'm not sure what the end result was but everything looks ok upon resumption.

**Comment by mzxyz** [comment]
Firstly, upgrade services will have no influence to the running projects.

A project can not announce `NOT INDEXING` with ongoing service agreements. In most cases when indexer upgrade the service to latest versions, the only impact is that the query services maybe down for a minute.

For prod projects, the better way is that indexer upgrade coordinator service separately which will have no impact for providing query services to the consumers, if need to upgrade proxy services or restart indexing project, will make the service down but the impact can be ignore for most cases. Also pull the upgrade images before run the docker-compose command also will save time.
